client/emv/emvcore.c

```diff
@@ -23,11 +94,370 @@ void TLVPrintFromBuffer(uint8_t *data, int datalen) {
 	struct tlvdb *t = NULL;
 	t = tlvdb_parse_multi(data, datalen);
 	if (t) {
-		PrintAndLog("TLV decoded:");
+		PrintAndLog("-------------------- TLV decoded --------------------");
 		
 		tlvdb_visit(t, print_cb, NULL, 0);
 		tlvdb_free(t);
 	} else {
 		PrintAndLog("TLV ERROR: Can't parse response as TLV tree.");
 	}
 }
+
+void TLVPrintFromTLV(struct tlvdb *tlv) {
+	if (!tlv) 
+		return;
+	
+	tlvdb_visit(tlv, print_cb, NULL, 0);
+}
+
+void TLVPrintAIDlistFromSelectTLV(struct tlvdb *tlv) {
+	PrintAndLog("|------------------|--------|-------------------------|");
+	PrintAndLog("|    AID           |Priority| Name                    |");
+	PrintAndLog("|------------------|--------|-------------------------|");
+
+	struct tlvdb *ttmp = tlvdb_find(tlv, 0x6f);
+	if (!ttmp)
+		PrintAndLog("|                         none                        |");
+		
+	while (ttmp) {
+		const struct tlv *tgAID = tlvdb_get_inchild(ttmp, 0x84, NULL);
+		const struct tlv *tgName = tlvdb_get_inchild(ttmp, 0x50, NULL);
+		const struct tlv *tgPrio = tlvdb_get_inchild(ttmp, 0x87, NULL);
+		if (!tgAID)
+			break;
+		PrintAndLog("|%s|   %s  |%s|", 
+			sprint_hex_inrow_ex(tgAID->value, tgAID->len, 18), 
+			(tgPrio) ? sprint_hex(tgPrio->value, 1) : "   ", 
+			(tgName) ? sprint_ascii_ex(tgName->value, tgName->len, 25) : "                         ");
+		
+		ttmp = tlvdb_find_next(ttmp, 0x6f);
+	}
+
+	PrintAndLog("|------------------|--------|-------------------------|");
+}
+
+
+int EMVSelect(bool ActivateField, bool LeaveFieldON, uint8_t *AID, size_t AIDLen, uint8_t *Result, size_t MaxResultLen, size_t *ResultLen, uint16_t *sw, struct tlvdb *tlv) {
+	uint8_t data[260] = {0};
```

Comment:
Here we have 260 again. Maybe there is a reason for it? E.g. 256 Bytes block size + 2 Response Status Bytes + 2 CRC Bytes added by iso14443a apdu?

Reply:
There was this reason, I have added more space to be sure.
There is one situation - chaining. In this situation APDU May be bigger. But now we dont support it on arm side and for emv it not needs. Cards with emv dont needs to support this sns ever simply not support.